docs/dev/core/internal/implementations.md

```diff
@@ -0,0 +1,17 @@
+# Implementations
+
+| Project Name      | Language Bindings   | Operating System  | App lifecycle type  | Environment Data source |
+| ----------------- | ------------------- | ----------------  | ------------------- | ----------------------- |
+| glean-core        | Rust                | all               | all                 | none                    |
+| glean-ffi         | C                   | all               | all                 | none                    |
+| glean-preview[^1] | Rust                | Windows/Mac/Linux | Desktop application | OS info build-time autodetected, app info passed in |
+| Glean Android     | Kotlin, Java        | Android           | Mobile app          | Autodetected from the Android environment |
+| Glean iOS         | Swift               | iOS               | Mobile app          | Autodetected from the iOS environment
+| Glean.py          | Python              | Windows/Mac/Linux | all                 | Autodetected at runtime |
```

Comment:
App lifecycle type of "all" here is fine for now -- we can fill in more detail as the app lifecycle type ping scheduling proposal progresses.